events/blocks/event-mailinglist-form/event-mailinglist-form.js

```diff
@@ -0,0 +1,169 @@
+  return emailPattern.test(input);
+}
+
+function decorateError(error, inputElement) {
+  inputElement.classList.add('error');
+  const errorSpan = inputElement.nextSibling;
+  errorSpan.classList.add('show');
+  errorSpan.innerHTML = error;
+}
+
+function decorateThankYouView(thanksView) {
+  const bp = {
+    thanksView,
+    thankyouTitle: thanksView.querySelector(':scope > div:nth-of-type(2) h2'),
+    thankyouDescription: thanksView.querySelector(':scope > div:nth-of-type(2) > p'),
+  };
+
+  bp.thankyouTitle.classList.add('thankyou-title');
+  bp.thankyouDescription.classList.add('thankyou-description');
+}
+
+async function subscribe(email) {
+  // If you have an API key, you can use it here.
+  return { email };
+  /* const response = await fetch('https://www.adobe.com/api2/subscribe_v1', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ email }),
+  });
+  return response.json();
+  */
+}
+
+async function handleSubmit(event, bp) {
+  event.preventDefault();
+
+  const inputElement = bp.block.querySelector(':scope > form > div > div:nth-of-type(2) > input');
+
+  // Validate the input
+  const email = inputElement.value;
+  if (email.length === 0) {
+    console.log('Email is a required Field');
+    decorateError('Required Field', inputElement);
+    return;
+  } if (!validateInput(email)) {
+    console.log('Input email must be a valid Email address');
+    decorateError('Must be a valid Email address', inputElement);
+    return;
+  }
+
+  try {
+    // Disable the multiple clicks.
+    event.target.disabled = true;
+
+    await subscribe(email);
+
+    // hide form view
+    const formView = bp.block.querySelector(':scope > form');
+
+    formView.classList.add('hide');
+
+    // show thankyou view
+    const thankyouView = bp.block.querySelector(':scope > div:nth-of-type(1)');
+    thankyouView.classList.remove('hide');
+    decorateThankYouView(thankyouView);
+  } catch (err) {
+    event.target.disabled = false;
+    console.error(err);
+    decorateError('Error subscribing', inputElement);
+  }
+}
+
+function decorateButton(bp) {
+  const button = createTag('button', { class: 'mailinglist-submit' });
+  button.innerHTML = bp.submitP.innerHTML;
+  button.addEventListener('click', (event) => {
+    handleSubmit(event, bp);
+  });
+  const div = createTag('div', { class: 'mailinglist-submit-container' });
+  bp.submitP.parentElement.appendChild(div);
+  div.appendChild(button);
+  bp.submitP.remove();
+}
+
+function addElementtoForm(form, inputP, labelP) {
+  const placeholder = inputP.innerHTML;
+  const labelText = labelP.innerHTML;
+  const labelAttr = {
+    for: 'email',
+    textContent: labelText,
+    class: 'mailinglist-label',
+  };
+
+  const inputAttr = {
+    type: 'email',
+    name: 'email',
+    placeholder,
+    required: 'true',
+    class: 'mailinglist-input',
+  };
+
+  const label = createTag('label', labelAttr);
+  label.innerHTML = labelText;
```

Comment:
Try to use `.textContent` instead of `.innerHTML` whenever possible.

Alternatively, for a lot of the `createTag` you can use the 3rd param to assign textContent and `{ parent: el-to-append-to }` in the 4th param to save quite a few lines! You might even be able to save some memory by not assigning the returned el to a variable.